src/postgres_deparse.c

```diff
@@ -130,173 +94,583 @@ isOp(const char *val)
-static void deparseMinMaxExpr(StringInfo str, MinMaxExpr *min_max_expr);
-static void deparseXmlExpr(StringInfo str, XmlExpr* xml_expr, DeparseNodeContext context);
-static void deparseXmlSerialize(StringInfo str, XmlSerialize *xml_serialize);
-static void deparseJsonIsPredicate(StringInfo str, JsonIsPredicate *json_is_predicate);
-static void deparseJsonObjectAgg(StringInfo str, JsonObjectAgg *json_object_agg);
-static void deparseJsonArrayAgg(StringInfo str, JsonArrayAgg *json_array_agg);
-static void deparseJsonObjectConstructor(StringInfo str, JsonObjectConstructor *json_object_constructor);
-static void deparseJsonArrayConstructor(StringInfo str, JsonArrayConstructor *json_array_constructor);
-static void deparseJsonArrayQueryConstructor(StringInfo str, JsonArrayQueryConstructor *json_array_query_constructor);
-static void deparseJsonValueExpr(StringInfo str, JsonValueExpr *json_value_expr);
-static void deparseJsonOutput(StringInfo str, JsonOutput *json_output);
-static void deparseJsonParseExpr(StringInfo str, JsonParseExpr *json_parse_expr);
-static void deparseJsonScalarExpr(StringInfo str, JsonScalarExpr *json_scalar_expr);
-static void deparseJsonSerializeExpr(StringInfo str, JsonSerializeExpr *json_serialize_expr);
-static void deparseJsonTable(StringInfo str, JsonTable *json_table);
-static void deparseJsonTableColumn(StringInfo str, JsonTableColumn *json_table_column);
-static void deparseJsonTableColumns(StringInfo str, List *json_table_columns);
-static void deparseJsonTablePathSpec(StringInfo str, JsonTablePathSpec *json_table_path_spec);
-static void deparseJsonBehavior(StringInfo str, JsonBehavior *json_behavior);
-static void deparseJsonFuncExpr(StringInfo str, JsonFuncExpr *json_func_expr);
-static void deparseJsonQuotesClauseOpt(StringInfo str, JsonQuotes quotes);
-static void deparseJsonOnErrorClauseOpt(StringInfo str, JsonBehavior *behavior);
-static void deparseJsonOnEmptyClauseOpt(StringInfo str, JsonBehavior *behavior);
-static void deparseConstraint(StringInfo str, Constraint *constraint);
-static void deparseSchemaStmt(StringInfo str, Node *node);
-static void deparseExecuteStmt(StringInfo str, ExecuteStmt *execute_stmt);
-static void deparseTriggerTransition(StringInfo str, TriggerTransition *trigger_transition);
-static void deparseCreateOpClassItem(StringInfo str, CreateOpClassItem *create_op_class_item);
-static void deparseAConst(StringInfo str, A_Const *a_const);
-static void deparseGroupingFunc(StringInfo str, GroupingFunc *grouping_func);
-
-static void deparsePreparableStmt(StringInfo str, Node *node);
-static void deparseRuleActionStmt(StringInfo str, Node *node);
-static void deparseExplainableStmt(StringInfo str, Node *node);
-static void deparseStmt(StringInfo str, Node *node);
-static void deparseValue(StringInfo str, union ValUnion *value, DeparseNodeContext context);
+// Each subpart is typically one line to be emitted in pretty print mode
+typedef struct DeparseStateSubPart
```

Comment:
What if we renamed this to `DeparseStatePart` and called the keyword-associated part list `DeparseStateGroup`? (or `DeparseStatePartGroup`)